src/client/components/localisation/skybox/view_model.ts

```diff
@@ -0,0 +1,88 @@
+import { createTransformer } from 'mobx'
+import { computed } from 'mobx'
+import { BackSide } from 'three'
+import { Mesh } from 'three'
+import { SphereBufferGeometry } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { PlaneBufferGeometry } from 'three'
+import { ShaderMaterial } from 'three'
+import { Vector3 } from 'three'
+import { SkyboxModel } from './model'
+import * as SkyboxFrag from './skybox.frag'
+import * as SkyboxVert from './skybox.vert'
+
+export class SkyboxViewModel {
+
+  public constructor(private model: SkyboxModel) {
+  }
+
+  public static of = createTransformer((model: SkyboxModel): SkyboxViewModel => {
+    return new SkyboxViewModel(model)
+  })
+
+  @computed
+  public get skybox() {
+    // reference: http://threejs.org/examples/#webgl_shaders_sky
+    const geo = new SphereBufferGeometry( 40, 32, 15 )
+    const mat = new ShaderMaterial({
+      fragmentShader: String(SkyboxFrag),
+      vertexShader: String(SkyboxVert),
+      uniforms: {
+        luminance: {value: this.model.luminance},
+        turbidity: {value: this.model.turbidity},
+        rayleigh: {value: this.model.rayleigh},
+        mieCoefficient: {value: this.model.mieCoefficient},
+        mieDirectionalG: {value: this.model.mieDirectionalG},
+        sunPosition: {value: this.sunPosition},
+      },
+      side: BackSide,
+    })
+
+    const mesh = new Mesh(geo, mat)
+    mesh.name = 'skyboxSky'
+
+    return mesh
+  }
+
+  @computed
+  public get ground() {
+    const groundGeo = new PlaneBufferGeometry( 40, 40 )
+    const groundMat = new MeshBasicMaterial( { color: 0x613610 } )
+    const ground = new Mesh( groundGeo, groundMat )
+    ground.name = 'skyboxGround'
+    ground.position.y = -0.01
+    ground.rotation.x = - Math.PI / 2
+
+    return ground
+  }
+
+  @computed
+  public get sun() {
+    const sunSphere = new Mesh(
+      new SphereBufferGeometry( 40, 16, 8 ),
+      new MeshBasicMaterial( { color: 0xffffff } ),
+    )
+
+    sunSphere.position.y = - 49
+    sunSphere.visible = false
+    sunSphere.name = 'skyboxSun'
+    sunSphere.position.copy(this.sunPosition)
+    sunSphere.visible = this.model.sun
```

Comment:
ah, instead of setting the visibility, we should just not add it to the scene if `model.sun` is false

i.e. the `skybox.add(this.skyboxSun)` call should be wrapped in an if:

```
if (this.model.sun) {
  skybox.add(this.skyboxSun)
}
```

in order to make this easier, I think we should move the 3 skybox properties directly into `SkyboxViewModel` so it can control this.

e.g. In `localisation/view_model`, we only have a single prop:

```
@computed
private get skybox() {
  return SkyboxViewModel.of(this.model.skybox).skybox
}
```

and then in `skybox/view_model`:

```
@computed
private get skybox() {
  const skybox = new Object3D()
  skybox.add(this.sky)
  skybox.add(this.ground)
  if (this.model.sun) {
    skybox.add(this.sun)
  }
  return skybox
}

@computed
private get sky() {
  ...
}
@computed
private get ground() {
  ...
}
@computed
private get sun() {
  ...
}
```